app/src/ui/blank-slate/create-tutorial-repository-dialog.tsx

```diff
@@ -114,12 +115,30 @@ export class CreateTutorialRepositoryDialog extends React.Component<
         true
       )
     } catch (err) {
-      throw new Error(
-        'Could not create the tutorial repository. ' +
-          'The most likely reason is that you already have a repository named ' +
-          `"${name}" on your account at ${friendlyEndpointName(account)}.\n\n` +
-          'Please delete the repository and try again.'
-      )
+      if (
+        err instanceof APIError &&
+        err.responseStatus === 422 &&
+        err.apiError !== null
+      ) {
+        if (err.apiError.message === 'Repository creation failed.') {
+          if (
```

Comment:
Could the 3 nested `if`s be combined into one large condition, or is it preferrable to keep them separate for readability?

Reply:
[non-blocking] Perhaps we can assume that we will only get the `name already exists on this account` message if repository creation failed, so we can drop the outer conditional and simply rely on the inner conditional check?

Reply:
Oh oops, got trigger happy and merged before giving @niik a chance to weigh in on this... Still curious to hear thoughts, and can make changes in a separate PR if we decide to